### PR TITLE
[release] fix long running horovod tune tests.

### PR DESCRIPTION
**Pull request by xwjiang2010**
Signed-off-by: xwjiang2010 <xwjiang2010@gmail.com>

## Related issue number
#29318

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by xwjiang2010** [comment]
tests running at https://buildkite.com/ray-project/release-tests-pr/builds/18646#0183f61c-7daf-4ee4-a3fd-e2345b16a02d

**Comment by xwjiang2010** [comment]
minor fix: https://buildkite.com/ray-project/release-tests-pr/builds/18674

**Comment by xwjiang2010** [comment]
Screen Shot 2022-10-20 at 12 24 27 PM

**Comment by xwjiang2010** [comment]
change to `device = ray.train.torch.get_device()`

**Comment by xwjiang2010** [comment]
final run: https://buildkite.com/ray-project/release-tests-pr/builds/18730

**Comment by xwjiang2010** [comment]
https://buildkite.com/ray-project/release-tests-pr/builds/18801

**Comment by xwjiang2010** [comment]
Both GPUs are being utilized to a fair amount. Memory is not creeping up.

Screen Shot 2022-10-21 at 10 52 19 AM
Screen Shot 2022-10-21 at 10 52 41 AM

File: python/ray/air/examples/horovod/horovod_tune_example.py [hunk 1]
```diff
@@ -52,7 +52,11 @@ def train_loop_per_worker(config):
     import horovod.torch as hvd
 
     hvd.init()
-    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    device = (
+        ray.train.torch.get_device()
+        if torch.cuda.is_available()
+        else torch.device("cpu")
+    )
```
Comment (amogkam) [inline python/ray/air/examples/horovod/horovod_tune_example.py:59]:
```suggestion
    device = ray.train.torch.get_device()
```

`ray.train.torch.get_device()` will work regardless of gpu or cpu.

File: release/air_tests/horovod/workloads/horovod_tune_test.py [hunk 2]
```diff
@@ -30,7 +38,11 @@ def train_loop_per_worker(config):
     import horovod.torch as hvd
 
     hvd.init()
-    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    device = (
+        ray.train.torch.get_device()
+        if torch.cuda.is_available()
+        else torch.device("cpu")
+    )
```
Comment (amogkam) [inline release/air_tests/horovod/workloads/horovod_tune_test.py:45]:
```suggestion
    device = ray.train.torch.get_device()
```

**Review by amogkam — APPROVED** [review]
Thanks @xwjiang2010!

**Comment by c21** [comment]
Hi @xwjiang2010 and @amogkam - does this PR need to be cherry picked to 2.1 release branch?

**Pull request merged**